tests/cross_functional/scale/noobaa/test_delete_objects.py

```diff
@@ -0,0 +1,190 @@
+import logging
+import pytest
+import time
+
+from threading import Event
+from concurrent.futures.thread import ThreadPoolExecutor
+from datetime import datetime
+from ocs_ci.ocs import hsbench
+from ocs_ci.ocs.bucket_utils import (
+    s3_delete_objects,
+    list_objects_in_batches,
+    s3_delete_object,
+    random_object_round_trip_verification,
+)
+from ocs_ci.framework.pytest_customization.marks import (
+    bugzilla,
+    polarion_id,
+    scale,
+    mcg,
+    orange_squad,
+)
+from ocs_ci.utility.retry import retry
+from ocs_ci.ocs.exceptions import CommandFailed
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="class")
+def s3bench(request):
+
+    s3bench = hsbench.HsBench()
+    s3bench.create_resource_hsbench()
+    s3bench.install_hsbench()
+
+    def teardown():
+        s3bench.cleanup()
+
+    request.addfinalizer(teardown)
+    return s3bench
+
+
+@scale
+@orange_squad
+@mcg
+class TestDeleteObjects:
+    def run_background_io(
+        self, io_pod, mcg_obj, bucket_name, upload_dir, download_dir, event
+    ):
+        """
+        Run background IO for the given bucket
+
+        Args:
+            io_pod (Pod): Pod object representing aws-cli pod
+            bucket_name (str): Name of the bucket
+            upload_dir (str): Pod directory from where the objects are uploaded
+            download_dir (str): Pod directory to where the objects are downloaded
+            event (threading.Event()): Event() object
+
+        """
+        while True:
+
+            retry(CommandFailed, tries=10, delay=5)(
+                random_object_round_trip_verification
+            )(
+                io_pod,
+                bucket_name,
+                upload_dir,
+                download_dir,
+                cleanup=True,
+                mcg_obj=mcg_obj,
+            )
+
+            if event.is_set():
+                break
+            time.sleep(60)
+        log.info(f"Successfully verified background io for the bucket {bucket_name}")
+
+    @bugzilla("2181535")
+    @polarion_id("OCS-4916")
+    @pytest.mark.parametrize(
+        argnames=["delete_mode"],
+        argvalues=[
+            # Below parameter is commented now because single deletion is causing
+            # heavy toll on memory consumption for the jenkins agent and test run time,
+            # as we have to delete one object at a time, till 2 million objects are deleted.
+            # pytest.param("single"),
```

Comment:
Delete_mode "single"  is not tested?

Reply:
We have tested and as the comment says it has heavy toll on memory consumption for the jenkins agent and also run time increases drastically. So we decided to omit this.